### PR TITLE
[REG-595]: Support functions for changing TLD ownership

### PR DESCRIPTION
**Pull request by altur-rang**

### CODE REVIEW

File: contracts/UNSRegistry.sol [hunk 1]
```diff
@@ -371,6 +371,33 @@ contract UNSRegistry is
...
     }
 
+    uint256 internal constant _ERC721_OWNERS_SLOT = 103;
+
+    function _moveTLDOwnership(uint256[] calldata tokenIds, address destination) internal onlyMintingManager {
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            require(ownerOf(tokenIds[i]) == address(0xdead), 'Registry: OWNER_NOT_0xDEAD');
+
+            bytes32 tokenSlot = keccak256(abi.encodePacked(tokenIds[i], _ERC721_OWNERS_SLOT));
+            StorageSlotUpgradeable.getAddressSlot(tokenSlot).value  = destination;
```
Comment (bogdan) [inline contracts/UNSRegistry.sol:381]: We need to emit `Transfer` event to make sure our mirror is consistent with changes.
Overall, I didn't understand why owner can not be changed via `_transfer`.

Comment (altur-rang) [reply contracts/UNSRegistry.sol:381]: Good point, we can use _transfer to transfer ownership to minting manager. But _transfer doesn't allow transfer to 0x0

Comment (altur-rang) [reply contracts/UNSRegistry.sol]: I have added use of _burn to burn tokens on L1. Thanks for idea!

File: contracts/MintingManager.sol [hunk 2]
```diff
@@ -217,6 +217,25 @@ contract MintingManager is ERC2771Context, MinterRole, Blocklist, Pausable, IMin
         unsRegistry.upgradeAll(tokenIds);
     }
 
+    function _isTldToChangeOwnership(uint256 tld) private view returns (bool) {
+        // 0x7674e7282552c15f203b9c4a6025aeaf28176ef7f5451b280f9bada3f8bc98e2 - .coin token ID
+        return bytes(_tlds[tld]).length > 0 || tld == 0x7674e7282552c15f203b9c4a6025aeaf28176ef7f5451b280f9bada3f8bc98e2;
```
Comment (nickshatilo) [inline contracts/MintingManager.sol:222]: Probably this value should be assigned to a constant

Comment (altur-rang) [reply contracts/MintingManager.sol:222]: Should we then use the constant in all places ?
https://github.com/unstoppabledomains/uns/blob/main/contracts/MintingManager.sol#L349

File: contracts/MintingManager.sol [hunk 3]
```diff
@@ -217,6 +217,25 @@ contract MintingManager is ERC2771Context, MinterRole, Blocklist, Pausable, IMin
...
+    }
+
+    function burnTLDL1(uint256[] calldata tokenIds) external onlyMinter {
```
Comment (aquiladev) [inline contracts/MintingManager.sol]:
```suggestion
    function burnTLDL1(uint256[] calldata tokenIds) external onlyOwner {
```

File: contracts/MintingManager.sol [hunk 4]
```diff
@@ -217,6 +217,25 @@ contract MintingManager is ERC2771Context, MinterRole, Blocklist, Pausable, IMin
...
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            require(_isTldToChangeOwnership(tokenIds[i]), 'MintingManager: TOKEN_ID_NOT_TLD');
+            unsRegistry.burnTLDL1(tokenIds[i]);
+       }
+    }
+
+    function moveTLDOwnershipL2(uint256[] calldata tokenIds) external onlyMinter {
```
Comment (aquiladev) [inline contracts/MintingManager.sol]:
```suggestion
    function moveTLDOwnershipL2(uint256[] calldata tokenIds) external onlyOwner {
```

**Review by bogdan — APPROVED** [review]

**Review by aquiladev — APPROVED** [review]

**Review by aquiladev — COMMENTED** [review]
We need to change minor version for MintingManager and UNSRegistry, because we gonna deploy these contracts

**Comment by altur-rang** [comment]
> We need to change minor version for MintingManager and UNSRegistry, because we gonna deploy these contracts

done

**Pull request merged**